src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -171,8 +181,10 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
                 volumes.put(ws, ws);
                 volumes.put(tmp, tmp);
             }
-
-            container = dockerClient.run(env, step.image, step.args, ws, volumes, volumesFromContainers, envReduced, dockerClient.whoAmI(), /* expected to hang until killed */ "cat");
+            if (step.entrypoint == null) {
+                step.entrypoint = "cat";
```

Comment:
See the [developer guide](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md#handling-default-values): better to initialize the field with the `"cat"` value, have the setter call `Util.fixEmpty`, and have `Image.exec` specify `entrypoint: ''`. Also be sure to adjust `WithContainerStep/config.jelly` to specify

```xml
<f:entry field="entrypoint" title="…">
    <f:textbox default="cat"/> <!-- or ${descriptor.defaultEntrypoint}, if defined -->
<f:entry>
```

and add `help-entrypoint.html`.

But anyway better would probably be to change this to a `boolean` field, since

* there is no use case for doing anything other than leaving the image along, or overriding its entrypoint to go to sleep
* there is a good reason to consider replacing `cat` with something like `sleep infinity` (except with a more POSIX-compliant argument)

The special techniques for step default values are the same for `boolean` parameters, but if you pick the semantics so that the default is in fact `false`, it is somewhat easier to write:

```xml
<f:checkbox/>
```

rather than

```xml
<f:checkbox default="true"/>
```

and a simpler form in the Java class.